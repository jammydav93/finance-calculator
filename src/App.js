import React from 'react';
import {
  Route,
  Switch,
  withRouter
} from 'react-router-dom';
//import SignUp from './components/SignUp';
import SignIn from './components/signin';
import Navigation from './components/navigation';
import Projector from './components/projector';
import './App.scss';
import withAuthentication from './components/withAuthentication';

let App = () => (
  <div className="app-container" >
    <h1 className="app-title">Finance Forecaster</h1>
    <Navigation />
    <div className="content">
      <Switch>
        <Route exact path="/" component={Projector} />
        <Route exact path="/signin" component={SignIn} />
        <Route exact path="/projector" component={Projector} />
      </Switch>
    </div>
  </div>
);

App = withAuthentication(App)

export default withRouter(App)
