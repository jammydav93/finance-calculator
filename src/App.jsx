import React from 'react';
import {
  Switch,
  Route,
} from 'react-router-dom';
import Navigation from './components/Navigation';
import Projector from './components/Projector';
import styles from './App.module.scss';
import WithAuthentication from './components/WithAuthentication';

export const App = () => (
  <div className={styles.app}>
    <div className={styles.header}>
      <Navigation />
    </div>
    <div className={styles.content}>
      <Switch>
        <Route path="/version">
          <div>{process.env.REACT_APP_VERSION}</div>
        </Route>
        <Route path="/">
          <Projector />
        </Route>
      </Switch>
    </div>
  </div>
);

const AuthenticatedApp = () => (
  <WithAuthentication>
    <App />
  </WithAuthentication>
);

export default AuthenticatedApp;
