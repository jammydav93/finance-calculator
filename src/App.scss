@mixin breakpoint($class) {
  @if $class == mobile {
    @media (max-width: 767px) { @content; }
  }
  @else if $class == tablet {
    @media (min-width: 768px) { @content; }
  }
  @else if $class == desktop {
    @media (min-width: 992px) { @content; }
  }
  @else {
    @warn "Breakpoint mixin supports: mobile, tablet, desktop";
  }
}

.app {
  text-align: center;
}

.app-header {
  background-color: #000;
  height: 150px;
  padding: 20px;
  color: white;
}

.app-title {
  text-align: center;
  font-size: 1.5em;
  @include breakpoint(mobile) {
    background: white;
  }
  @include breakpoint(tablet) {
    background: pink;
  }
  @include breakpoint(desktop) {
    background: yellow;
  }
}

body {
  background-color: #a0c1f7;
  padding: 20px;
  min-width: 320px;
  font-family: sans-serif;
  @include breakpoint(mobile) {
    padding: 20px 0px;
  }
}

.content {
  background-color: #FFF;
  margin: 0 auto;
}

.active {
  background-color: #0099FF;
}