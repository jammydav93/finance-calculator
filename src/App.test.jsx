import React from 'react';
import { shallow } from 'enzyme';
import { App } from './App';

it('App snapshots match', () => {
  const component = shallow(<App />);

  expect(component).toMatchSnapshot();
});
