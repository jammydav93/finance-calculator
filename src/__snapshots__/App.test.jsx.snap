// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App snapshots match 1`] = `ShallowWrapper {}`;
